.github/workflows/solidity.yml

```
@@ -0,0 +1,124 @@
+
+      - name: Build contracts
+        run: forge build
+
+      - name: Cache data
+        uses: actions/cache@v4
+        with:
+          path: contracts
+          key: ci-solidity-${{ github.ref }}
+
+      - id: forge
+        run: echo "FORGE_PATH=$(which forge)" >> $GITHUB_OUTPUT
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:37: Double quote to prevent globbing and word splitting

(shellcheck)